zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/archiver/ReschedulingTaskTest.java

```diff
@@ -0,0 +1,158 @@
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 12L, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  void shouldResetErrorDelayOnSuccessfulArchiving() {
+    // given
+    final var job =
+        new ArchiverJob() {
+          private final AtomicInteger count = new AtomicInteger();
+
+          @Override
+          public CompletableFuture<Integer> archiveNextBatch() {
+            final var runCount = count.getAndIncrement();
+            if (runCount == 0) {
+              return CompletableFuture.failedFuture(new RuntimeException("error"));
+            }
+
+            if (runCount == 2) {
+              return CompletableFuture.failedFuture(new RuntimeException("error"));
+            }
+
+            return CompletableFuture.completedFuture(1);
+          }
+        };
+    final var task = new ReschedulingTask(job, 1, 10L, executor, LOGGER);
+
+    // when
+    task.run();
+
+    // then - scheduled after the minimum delay on the first error, again on success, and then
+    // since we reset the error delay, again on the second error
+    final var inOrder = Mockito.inOrder(executor);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 10L, TimeUnit.MILLISECONDS);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 10L, TimeUnit.MILLISECONDS);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 10L, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  void shouldRescheduleEvenIfJobResultIsNull() {
+    // given
+    final var job =
+        new ArchiverJob() {
+          @Override
+          public CompletableFuture<Integer> archiveNextBatch() {
+            return null;
+          }
+        };
+    final var task = new ReschedulingTask(job, 1, 10L, executor, LOGGER);
+
+    // when
+    task.run();
+
+    // then
+    final var inOrder = Mockito.inOrder(executor);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 10L, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  void shouldRescheduleTaskWithDelayOnPartialBatch() {
+    // given
+    final var job =
+        new ArchiverJob() {
+          @Override
+          public CompletableFuture<Integer> archiveNextBatch() {
+            return CompletableFuture.completedFuture(1);
+          }
+        };
+    final var task = new ReschedulingTask(job, 2, 10L, executor, LOGGER);
+
+    // when
+    task.run();
+
+    // then
+    final var inOrder = Mockito.inOrder(executor);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 10L, TimeUnit.MILLISECONDS);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 12L, TimeUnit.MILLISECONDS);
+    inOrder
+        .verify(executor, Mockito.timeout(5_000).times(1))
+        .schedule(task, 14L, TimeUnit.MILLISECONDS);
```

Comment:
Good point! Either this, or a comment explaining it. The main reason for such small delays is to keep the test fast, since we still wait for the delay between scheduling. But I'll add a comment :+1:

Reply:
I commented it in the next PR to avoid rebasing both :sweat_smile: